scripts/terraform/aws/m3-db/main.tf

```diff
@@ -0,0 +1,67 @@
+# durable network interfaces
+resource "aws_network_interface" "m3-db" {
+  subnet_id       = "${var.subnet_id}"
+  security_groups = ["${var.sg_ids}"]
+
+  tags {
+    Application          = "m3"
+    Name                 = "${var.instance_name}"
+    configuration-source = "${var.config_source}"
+  }
+}
+
+# public elastic IPs
+resource "aws_eip" "m3-db" {
```

Comment:
Is the need for public IP to have m3db nodes communicate regionally?

Reply:
Good question and the answer is no.

The reason it's really here is that this will also persist the private IP between when you destroy/terminate and instance and when you recreate it. The docs recommend that you keep the hostname and IP address the same for simplicity sake. I can update it so it will have a private IP address that is "sticky".